tests/tests_fabric/strategies/test_deepspeed_integration.py

```diff
@@ -31,78 +31,79 @@
-        model_1 = BoringModel()
-        optimizer_1 = torch.optim.SGD(model_1.parameters(), lr=0.0001)
-
-        fabric_obj.seed_everything(42)
-        model_2 = BoringModel()
-        optimizer_2 = torch.optim.SGD(model_2.parameters(), lr=0.0001)
-
-        for mw_1, mw_2 in zip(model_1.state_dict().values(), model_2.state_dict().values()):
-            assert torch.allclose(mw_1, mw_2)
-
-        model_1, optimizer_1 = fabric_obj.setup(model_1, optimizer_1)
-        model_2, optimizer_2 = fabric_obj.setup(model_2, optimizer_2)
-
-        # train model_1 first
-        fabric_obj.seed_everything(42)
-        data_list = []
-        for _ in range(2):
-            optimizer_1.zero_grad()
-            data = torch.randn(1, 32).to(fabric_obj.device)
-            data_list.append(data)
-            x = model_1(data)
-            loss = x.sum()
-            fabric_obj.backward(loss, model=model_1)
-            optimizer_1.step()
-
-        # the weights do not match
-        assert all(w.nelement() > 1 for w in model_1.state_dict().values())
-        assert all(w.nelement() == 0 for w in model_2.state_dict().values())
-
-        # now train model_2 with the same data
-        for data in data_list:
-            optimizer_2.zero_grad()
-            x = model_2(data)
-            loss = x.sum()
-            fabric_obj.backward(loss, model=model_2)
-            optimizer_2.step()
-
-        # the weights should match
-        for mw_1, mw_2 in zip(model_1.state_dict().values(), model_2.state_dict().values()):
-            assert torch.allclose(mw_1, mw_2)
-
-        # Verify collectives works as expected
-        ranks = fabric_obj.all_gather(torch.tensor([fabric_obj.local_rank]).to(fabric_obj.device))
-        assert torch.allclose(ranks.cpu(), torch.tensor([[0], [1]]))
-        assert fabric_obj.broadcast(True)
-        assert fabric_obj.is_global_zero == (fabric_obj.local_rank == 0)
 
-    fabric = Fabric(strategy=DeepSpeedStrategy(stage=3, logging_batch_size_per_gpu=1), devices=2, accelerator="gpu")
-    fabric.launch(run)
+    optimizer = torch.optim.SGD(model.parameters(), lr=0.0001)
+    model, optimizer = fabric.setup(model, optimizer)
+
+    for i in range(2):
+        optimizer.zero_grad()
+        x = model(torch.randn(1, 32).to(fabric.device))
+        loss = x.sum()
+        if i == 0:
+            # the weights are not initialized with stage 3 until backward is run once
+            assert all(w.nelement() == 0 for w in model.state_dict().values())
+        fabric.backward(loss, model=model)
+        if i == 0:
+            # save for later to check that the weights were updated
+            state_dict = deepcopy(model.state_dict())
+        optimizer.step()
+
+    # check that the model trained, the weights from step 1 do not match the weights from step 2
+    for mw_b, mw_a in zip(state_dict.values(), model.state_dict().values()):
+        assert not torch.allclose(mw_b, mw_a)
+
+    fabric.seed_everything(42)
```

Comment:
unrelated note to this PR, but shall we have seed as a wrapper to be called `with seed_...` so we can have it with model

Reply:
In all cases I know, it is desired to set the seed globally for the program, but if you find a use case for it to be set locally only, feel free to open an issue or PR :)